api/v1beta1/gcpmachine_types.go

```diff
@@ -47,7 +47,10 @@ type AttachedDiskSpec struct {
 	// 1. "pd-standard" - Standard (HDD) persistent disk
 	// 2. "pd-ssd" - SSD persistent disk
 	// 3. "local-ssd" - Local SSD disk (https://cloud.google.com/compute/docs/disks/local-ssd).
+	// 4. "pd-balanced" - Balanced Persistent Disk
+	// 5. "hyperdisk-balanced" - Hyperdisk Balanced
 	// Default is "pd-standard".
+	// +kubebuilder:validation:Enum=pd-standard;pd-ssd;local-ssd;pd-balanced;hyperdisk-balanced
```

Comment:
I assumed that there would be some existing validation to extend, but it looks like there isn't.
This new validation would be a breaking change, since users may already have content with values that do not match this.

In theory you could use a CEL expression here to write some ratcheting validation, but I'm not sure if the CAPI maintainers are ok with CEL yet.

Alternatively, since there are webhooks, you could write your own ratcheting validation there, which allows the value on update, but doesn't allow on create. It would still be a PITA for people though if somehow they hit this while a MachineSet was creating a new Machine

Reply:
I know there are discussions about CEL at the core CAPI level, but personally, I see no problems with us using it in CAPG.

As you say, new validation would require an API version bump. I have created #1287 to track API version changes we want.

Reply:
@richardcase Is it acceptable if we go back to the original change and just add to the comments. We don't want to add a breaking change if we don't need to.

Reply:
It would be acceptable.

Would you also be able to add a comment to #1287 so that we make sure we can add the validation rule / CEL when we bump the version?

Reply:
@richardcase I changed this back to the original diff. I also commented on #1287. 

PTAL and Thank you.

Reply:
@cpanato PTAL thank you.